staged-documentation.md

```diff
@@ -0,0 +1,282 @@
+## Overriding bean factories
+
+When a bean does not contain all the data required for the template, you can override its factory to produce a subclass of the bean.
+
+In this example shows how to add a field `lastUpdated` to the `ProductDetailPageContent`. This procedure is not limited to page class beans.
+
+First override the bean to add the field including getters and setters:
+
+```java
+public class MyProductDetailPageContent extends ProductDetailPageContent {
+    private String lastUpdated;
+
+    public String getLastUpdated() {
+        return lastUpdated;
+    }
+
+    public void setLastUpdated(final String lastUpdated) {
+        this.lastUpdated = lastUpdated;
+    }
+}
+```
+
+Then override the factory for the original bean:
+
+```java
+public class MyProductDetailPageContentFactory extends ProductDetailPageContentFactory {
+    //here you can use dependency injection to get instances
+    @Inject
+    private UserContext userContext;
+
+    //do not call here super.create because you would get a bean of the original class and not the subclass
+    @Override
+    public ProductDetailPageContent create(final ProductProjection product, final ProductVariant variant) {
+        final MyProductDetailPageContent bean = new MyProductDetailPageContent();
+        initialize(bean, product, variant);
+        return bean;
+    }
+
+    //the new initialize method contains as first parameter the bean of your subclass and then the parameters of the create method
+    protected final void initialize(final MyProductDetailPageContent bean, final ProductProjection product, final ProductVariant variant) {
```

Comment:
No, I think the comments help a lot. I would simply avoid creating this `initialize` method, so that they don't have to be careful not to override the `super.initialize`or call the same method infinitely. If you agree, I can change it myself.